src/components/FillArrow.tsx

```diff
@@ -1,13 +1,48 @@
+import { useRef, useEffect, useMemo, useState } from 'react';
 import '../styles/FillArrow.scss';
+import { VscDebugRestart } from 'react-icons/vsc';
 
 interface FillArrowProps {
   text1: string;
   text2: string;
 }
 
+function useIsInViewport(ref) {
+  const [isIntersecting, setIsIntersecting] = useState(false);
+
+  const observer = useMemo(
+    () =>
+      new IntersectionObserver(([entry]) =>
+        setIsIntersecting(entry.isIntersecting)
+      ),
+    []
+  );
+
+  useEffect(() => {
+    observer.observe(ref.current);
+
+    return () => {
+      observer.disconnect();
+    };
+  }, [ref, observer]);
+
+  return isIntersecting;
+}
+
 function FillArrow(props: FillArrowProps): JSX.Element {
+  const ref = useRef(null);
+
+  const isInViewport = useIsInViewport(ref);
+
+  const [animate, setAnimate] = useState(false);
+
+  useEffect(() => {
+    if (isInViewport && !animate) setAnimate(true);
+  }, [isInViewport, animate]);
+
   return (
-    <>
+    <div ref={ref} className="overall-container">
+      <VscDebugRestart className="reset" onClick={() => setAnimate(false)} />
```

Comment:
Let's remove this button because its kind of buggy and doesn't really feel good to click. Let's just have the animation loop automatically instead, like a GIF.

Reply:
can't seem to animate it like a gif but i removed the button. Probably will look into it at another date

Reply:
Ok, I can take a look as well, but I will ticket.